src/main/java/io/lettuce/core/StatefulRedisConnectionImpl.java

```diff
@@ -273,12 +308,20 @@ protected <T> RedisCommand<K, V, T> preProcessCommand(RedisCommand<K, V, T> comm
 
         if (commandType.equals(MULTI.name())) {
 
+            reAuthSafety.lock();
+            try {
+                inTransaction.set(true);
+            } finally {
+                reAuthSafety.unlock();
+            }
             multi = (multi == null ? new MultiOutput<>(codec) : multi);
 
             if (command instanceof CompleteableCommand) {
                 ((CompleteableCommand<?>) command).onComplete((ignored, e) -> {
                     if (e != null) {
                         multi = null;
+                        inTransaction.set(false);
```

Comment:
Looking at the code you can provide directly [Command](https://github.com/ggivo/lettuce/blob/b91f78590cc97b6f5126ded053043ac5bfb9abcf/src/main/java/io/lettuce/core/protocol/Command.java#L37) to dispath. 
What is the actual use? I am not sure but if it is not `CompletableCommand`  it will make most of `preProcessCommand() `method irrelevant.
Not sure if @tishun  can add here.

Reply:
🤷 

I thought it was that all sync commands are not completable commands.
But why are they exempt from these checks I am not sure.

Does MULTI even work with sync commands?